packages/patrol_cli/lib/src/features/test/android_test_backend.dart

```diff
@@ -1,41 +1,73 @@
 import 'dart:convert' show base64Encode, utf8;
+import 'dart:io' show Process;
 
+import 'package:adb/adb.dart';
 import 'package:dispose_scope/dispose_scope.dart';
 import 'package:file/file.dart';
 import 'package:path/path.dart' show basename;
 import 'package:patrol_cli/src/common/extensions/process.dart';
 import 'package:patrol_cli/src/common/logger.dart';
 import 'package:patrol_cli/src/features/run_commons/device.dart';
+import 'package:patrol_cli/src/features/test/test_backend.dart';
 import 'package:platform/platform.dart';
 import 'package:process/process.dart';
 
-class AndroidAppOptions {
+class AndroidAppOptions extends AppOptions {
   const AndroidAppOptions({
-    required this.target,
-    required this.flavor,
-    required this.dartDefines,
+    required super.target,
+    required super.flavor,
+    required super.dartDefines,
   });
 
-  final String target;
-  final String? flavor;
-  final Map<String, String> dartDefines;
+  @override
+  String get desc => 'apk with entrypoint ${basename(target)}';
+
+  List<String> toGradleAssembleInvocation({required bool isWindows}) {
+    final flavor = _effectiveFlavor(this.flavor);
+    return _toGradleInvocation(
+      isWindows: isWindows,
+      task: 'assemble${flavor}Debug',
+    );
+  }
+
+  List<String> toGradleAssembleTestInvocation({required bool isWindows}) {
+    final flavor = _effectiveFlavor(this.flavor);
+    return _toGradleInvocation(
+      isWindows: isWindows,
+      task: 'assemble${flavor}DebugAndroidTest',
+    );
+  }
+
+  List<String> toGradleConnectedTestInvocation({required bool isWindows}) {
+    final flavor = _effectiveFlavor(this.flavor);
+    return _toGradleInvocation(
+      isWindows: isWindows,
+      task: 'connected${flavor}DebugAndroidTest',
+    );
+  }
+
+  String _effectiveFlavor(String? flavor) {
```

Comment:
Ah you're not even using the `flavor` method parameter. Let's make this a getter

Reply:
Nice catch